include/odp/api/spec/packet_io_stats.h

```diff
@@ -163,8 +206,75 @@ typedef struct odp_pktio_stats_capability_t {
 		};
 	} pktio;
 
+	/** Input queue level capabilities */
+	struct {
+		/** Supported counters */
+		union {
+			/** Statistics counters in a bit field structure */
+			struct {
+				/** @see odp_pktin_queue_stats_t::in_octets */
+				uint64_t in_octets          : 1;
+
+				/** @see odp_pktin_queue_stats_t::in_packets */
+				uint64_t in_packets         : 1;
+
+				/** @see odp_pktin_queue_stats_t::in_discards */
+				uint64_t in_discards        : 1;
+
+				/** @see odp_pktin_queue_stats_t::in_errors */
+				uint64_t in_errors          : 1;
+			} counter;
+
+			/** All bits of the bit field structure
+			 *
+			 *  This field can be used to set/clear all flags, or
+			 *  for bitwise operations over the entire structure. */
+			uint64_t all_counters;
+		};
+	} pktin_queue;
+
+	/** Output queue level capabilities */
+	struct {
+		/** Supported counters */
+		union {
+			/** Statistics counters in a bit field structure */
+			struct {
+				/** @see odp_pktout_queue_stats_t::out_octets */
+				uint64_t out_octets         : 1;
+
+				/** @see odp_pktout_queue_stats_t::out_packets */
+				uint64_t out_packets        : 1;
+
+				/** @see odp_pktout_queue_stats_t::out_discards */
+				uint64_t out_discards       : 1;
+
+				/** @see odp_pktout_queue_stats_t::out_errors */
+				uint64_t out_errors         : 1;
+			} counter;
+
+			/** All bits of the bit field structure
+			 *
+			 *  This field can be used to set/clear all flags, or
+			 *  for bitwise operations over the entire structure. */
+			uint64_t all_counters;
+		};
+	} pktout_queue;
+
 } odp_pktio_stats_capability_t;
 
+/**
+ * Union of all queue handle types used for packet input/output
+ */
+typedef union {
+	/** Event queue handle for packet input or output */
+	odp_queue_t event_queue;
+	/** Direct packet input queue handle */
+	odp_pktin_queue_t pktin_queue;
+	/** Direct packet output queue handle */
+	odp_pktout_queue_t pktout_queue;
```

Comment:
Need TM queue and CoS handle also. Right?

Reply:
That was not the intention. Concept-wise these are so different compared to pktio that I would prefer adding separate statistics functions for these two modules.

Reply:
Ok. Makes sense to add as separate functions. Will it be in this patch or another patch in series?

Reply:
They'll be in a separate PR.